xgcm/transform.py

```diff
@@ -117,18 +117,28 @@ def _interp_1d_conservative(phi, theta_1, theta_2, theta_hat_1, theta_hat_2, out
             theta_max = theta_1[i]
 
         for j in range(m):
+            if np.isnan(phi[i]):
+                # missing values in phi (the input data) need to be excluded
+                # here or the whole target bin will be NaN
+                continue
             if (theta_hat_1[j] > theta_max) or (theta_hat_2[j] < theta_min):
                 # there is no overlap between the cell and the bin
                 pass
             elif theta_max == theta_min:
-                output[j] += phi[i]
+                if np.isnan(output[j]):
+                    output[j] = phi[i]
+                else:
+                    output[j] += phi[i]
```

Comment:
I am unsure how much of a performance hit we might get from this additional if/else statement...